fix(entity-feedback): Add link to feedback notifications

west-david opened the pull request:
## Summary

This PR fixes a UX issue in the entity-feedback-backend plugin where notifications about new feedback don't include a clickable link to navigate to the entity.

## Changes

- Added `link` field to notification payload in `/responses/:entityRef(*)` endpoint
- Link format: `/catalog/{namespace}/{kind}/{name}/feedback`
- Parses entityRef (format: `kind:namespace/name`) to construct proper Backstage entity URL

## Motivation

When entity owners receive notifications about new feedback on their entities, the notification currently shows the entity name and comments but doesn't provide a way to navigate to the feedback. This PR adds a clickable link that takes users directly to the entity's feedback page.

## Implementation

```typescript
// Construct entity URL from entityRef (format: "kind:namespace/name")
const [kind, namespaceName] = req.params.entityRef.split(":");
const [namespace, name] = namespaceName.split("/");
const entityUrl = `/catalog/${namespace}/${kind}/${name}/feedback`;

const payload: NotificationPayload = {
  title: `New feedback for ${req.params.entityRef}`,
  description: `Comments: ${JSON.parse(comments).additionalComments}`,
  link: entityUrl,
};
```

## Testing

Tested in a production Backstage instance with entity-feedback plugin enabled:
1. Submitted feedback on a test entity
2. Verified notification was sent to entity owner
3. Confirmed notification includes clickable link
4. Verified link navigates correctly to entity's feedback page

The notification link successfully takes users to the correct entity feedback page in the format `/catalog/{namespace}/{kind}/{name}/feedback`.

kuangp commented on workspaces/entity-feedback/plugins/entity-feedback-backend/src/service/router.ts:
```diff
@@ -227,9 +227,16 @@ export async function createRouter(
+      // Construct entity URL from entityRef (format: "kind:namespace/name")
+      const [kind, namespaceName] = req.params.entityRef.split(':');
+      const [namespace, name] = namespaceName.split('/');
+      const entityUrl = `/catalog/${namespace}/${kind}/${name}/feedback`;
```
Thanks for suggesting this change however it is not guaranteed that this is always the entity URL - consumers are free to attach the catalog plugin and "feedback" tab to any custom route name of their choosing even though this is the default. Therefore we probably need a more configurable way to construct this url that is consistent with how the frontend plugin is setup

west-david commented:
Thanks for the review! I've addressed both comments:

**1. Using `parseEntityRef`**: ✅ 
Replaced manual string parsing with the `parseEntityRef` utility from `@backstage/catalog-model`.

**2. Configurable URL pattern**: ✅ 
Made the notification link URL configurable via app-config to support custom routes:

```typescript
const feedbackUrlPattern =
  config.getOptionalString('entityFeedback.feedbackUrlPattern') ||
  '/catalog/:namespace/:kind/:name';

const entityUrl = feedbackUrlPattern
  .replace(':kind', kind)
  .replace(':namespace', namespace)
  .replace(':name', name);
```

**Default behavior**: Links to `/catalog/:namespace/:kind/:name` (without `/feedback` suffix)

**Custom configuration example**:
```yaml
entityFeedback:
  feedbackUrlPattern: '/catalog/:namespace/:kind/:name/feedback'
```

This allows users to configure the exact pattern that matches their frontend routing setup.

Tests have been updated to reflect the new default pattern.

kuangp commented on workspaces/entity-feedback/plugins/entity-feedback-backend/src/service/router.ts:
```diff
@@ -227,9 +228,25 @@ export async function createRouter(
+      // Allow customization of the feedback URL pattern via config
+      // Default: /catalog/:namespace/:kind/:name
+      const feedbackUrlPattern =
+        config.getOptionalString('entityFeedback.feedbackUrlPattern') ||
```
actually another option which may be more robust is to send this info from the frontend plugin since the routing can be derived there automatically: 

if you can emulate what's done here to derive the route of the entity: https://github.com/backstage/backstage/blob/fea3e3972daf42fc75c107568a693eeb14a4d672/plugins/catalog-react/src/components/EntityRefLink/EntityRefLink.tsx#L84-L88, then pass it along in the request to the backend here: https://github.com/backstage/community-plugins/blob/4d58c3f3004d7bf7aa8c1436954eb1189df5c58d/workspaces/entity-feedback/plugins/entity-feedback/src/components/FeedbackResponseDialog/FeedbackResponseDialog.tsx#L132, it would be available in this express route for it to be included in the notification payload

The benefit of the above is that it will always consistently construct the appropriate entity url (including custom ones) without any additional configuration from the consumer

kuangp reviewed (approved):
Looks great! Thanks !
